FWCore/Framework/interface/EDConsumerBase.h

```diff
@@ -97,7 +99,9 @@ namespace edm {
     typedef ProductLabels Labels;
     void labelsForToken(EDGetToken iToken, Labels& oLabels) const;
 
-    void modulesWhoseProductsAreConsumed(std::vector<ModuleDescription const*>& modules,
+    void modulesWhoseProductsAreConsumed(std::vector<ModuleDescription const*>& modulesEvent,
+                                         std::vector<ModuleDescription const*>& modulesLumiRun,
```

Comment:
In light of `ProcessBlock` I'm thinking this parameter (and the corresponding storage in `PathsAndConsumesOfModules`) be either renamed to `modulesNonEvent`, or separate the `Lumi`, `Run`, and (eventually) `Process`.

Reply:
Would it be better to do `std::array<std::vector<ModuleDescription const*>, Transition::NumberOfTransitions>` which would allow us to add transitions without changing the interface again?

Reply:
It would, thanks for the suggestion.

Reply:
Would the transition be the transition in which the product is consumed or the transition in which the product is produced? Are we adding the transition to the interface of the consumes function or where does this information come from?

Reply:
Thanks @wddgit. It is actually the branch type where the product exists, so the parameter should be `std::array<std::vector<ModuleDescription const*>, NumBranchTypes>`.

Reply:
Changed to `std::array<std::vector<ModuleDescription const*>* , NumBranchTypes>` (need an array of pointers because they are stored as `std::vector<std::vector<ModuleDescription>>` and arrays of references are not allowed).